### PR TITLE
Benchmarking Ray Data bulk ingest as input file size changes.

### PR DESCRIPTION
**Pull request by cadedaniel**
This PR adds a benchmark which takes work from https://github.com/anyscale/air-benchmarks and makes it run as a release test.

Full metrics are stored in Databricks:
```
[
    {
        "time_total_s": 66.81319880485535,
        "ray_spilled_mb": 0,
        "ray_restored_mb": 0,
        "min_available_mb": 34939.1328125,
        "ray_mem_monitor_enabled": "True",
        "data_loader": "ray.data",
        "train_sleep_time_ms": 0,
        "num_epochs": 1,
        "num_images_per_epoch": 20480,
        "num_images_per_input_file": 32,
        "batch_size": 64,
        "shuffle_buffer_size": 0,
        "tput_images_per_s": 306.5262607739671
    },
    {
        "time_total_s": 63.11659026145935,
        "ray_spilled_mb": 0,
        "ray_restored_mb": 0,
        "min_available_mb": 26121.12109375,
        "ray_mem_monitor_enabled": "True",
        "data_loader": "ray.data",
        "train_sleep_time_ms": 0,
        "num_epochs": 1,
        "num_images_per_epoch": 20480,
        "num_images_per_input_file": 512,
        "batch_size": 64,
        "shuffle_buffer_size": 0,
        "tput_images_per_s": 324.4788718015654
    },
    {
        "time_total_s": 117.54973983764648,
        "ray_spilled_mb": 0,
        "ray_restored_mb": 0,
        "min_available_mb": 5155.19921875,
        "ray_mem_monitor_enabled": "True",
        "data_loader": "ray.data",
        "train_sleep_time_ms": 0,
        "num_epochs": 1,
        "num_images_per_epoch": 20480,
        "num_images_per_input_file": 2048,
        "batch_size": 64,
        "shuffle_buffer_size": 0,
        "tput_images_per_s": 174.22412017488
    }
]
```

Performance metrics are exported:
```
[
    {
        "perf_metric_name": "ray.data_32-images-per-file_throughput-img-per-second",
        "perf_metric_value": 306.5262607739671,
        "perf_metric_type": "THROUGHPUT"
    },
    {
        "perf_metric_name": "ray.data_512-images-per-file_throughput-img-per-second",
        "perf_metric_value": 324.4788718015654,
        "perf_metric_type": "THROUGHPUT"
    },
    {
        "perf_metric_name": "ray.data_2048-images-per-file_throughput-img-per-second",
        "perf_metric_value": 174.22412017488,
        "perf_metric_type": "THROUGHPUT"
    }
]
```

## Testing
I ran the release test from BK: https://buildkite.com/ray-project/release-tests-pr/builds/18592

Locally, I tested via the following command:
```
ANYSCALE_HOST=https://console.anyscale-staging.com \
  ANYSCALE_CLI_TOKEN=xyz \
  PYTHONPATH=$PYTHONPATH:$(pwd) \
  python3 ray_release/scripts/run_release_test.py ray-data-bulk-ingest-file-size-benchmark \
  --ray-wheels master \
  --no-terminate \
```

## Dashboard screenshots
Screen Shot 2022-10-19 at 3 23 35 PM
Screen Shot 2022-10-19 at 3 23 42 PM

### CODE REVIEW

File: release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+    
+    num_images_per_file = metrics["num_images_per_input_file"]
+    data_loader = metrics["data_loader"]
+
+    perf_metrics = output_json.get("perf_metrics", [])
+    perf_metrics.append({
+        "perf_metric_name": f"{data_loader}_{num_images_per_file}-images-per-file_throughput-img-per-second",
+        "perf_metric_value": metrics["tput_images_per_s"],
+        "perf_metric_type": "THROUGHPUT",
+    })
```
Comment (stephanie-wang) [inline release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py:362]: @jjyao should confirm but I think you may also have to set "time" and "success" as top-level JSON fields.

For "success", probably best to check whether an exception was thrown in the main training loop (`exc is not None`).

**Review by stephanie-wang — APPROVED** [review]

File: release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py (hunk 1 above)
Comment (jjyao) [reply release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py:371]: Those are not required.

**Comment by jjyao** [comment]
@cadedaniel could you trigger a run and paste the buildkite link?

File: release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py (hunk 1 above)
Comment (stephanie-wang) [reply release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py]: Ah okay I thought success would get reported to the nightly dashboard, but I guess not! Yeah it's fine if we don't add duration.

**Pull request merged**